types/index.d.ts

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+// Minimum TypeScript Version: 3.0
+
+declare namespace StyleDictionary {
+  interface Property {
+    value: string;
+    [attribute: string]: any;
```

Comment:
I've been away from typescript for awhile. I think that one goal is to avoid the use of 'any'.

Is it possible that this should be "string"? Or does it make sense in this context because this is how SD is structured?

If you have pointers to the logic here would be helpful for me to validate.

Reply:
Yes it is best practise to avoid `any`, however I used it here as the config JSON can contain any additional attribute. Thinking about this a bit more perhaps it would be better scoped to the types valid in JSON?
Perhaps something like this: 
```typescript
type JSONValue = null | string | JSONValue[] | JSONObject;
interface JSONObject {
    [key: string]: JSONValue;
}
```

Reply:
Also looking at this https://github.com/AndrewLeedham/style-dictionary/blob/master/lib/transform/propertySetup.js#L35 I think the internal type I called `Prop` should also have this behaviour.

Reply:
The `JSONObject` idea is not feasible. Unfortunately because other properties exist on the interfaces, an index signature would have to include non-valid props as well.

The built in `JSON.parse` function returns `any` so I think we will have to use `any` here.